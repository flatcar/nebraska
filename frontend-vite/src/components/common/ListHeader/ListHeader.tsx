import { Box } from '@mui/material';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import makeStyles from '@mui/styles/makeStyles';
import React from 'react';

const useStyles = makeStyles({
  sectionHeader: {
    padding: '1em',
  },
});

export default function ListHeader(props: { title: string; actions?: React.ReactElement[] }) {
  const classes = useStyles();
  const actions = props.actions || [];
  return (
    <Grid
      container
      alignItems="flex-start"
      justifyContent="space-between"
      className={classes.sectionHeader}
    >
      {props.title && (
        <Grid item>
          <Box p={0.5}>
            <Typography variant="h1">{props.title}</Typography>
          </Box>
        </Grid>
      )}
      {actions &&
        actions.map((action, i: number) => (
          <Grid item key={i}>
            {action}
          </Grid>
        ))}
    </Grid>
  );
}
