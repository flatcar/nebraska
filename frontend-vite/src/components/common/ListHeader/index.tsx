export { default } from './ListHeader';
export * from './ListHeader';
