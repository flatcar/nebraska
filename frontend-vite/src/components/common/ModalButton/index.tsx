export { default } from './ModalButton';
export * from './ModalButton';
