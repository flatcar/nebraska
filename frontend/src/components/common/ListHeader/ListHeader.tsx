import { Box } from '@mui/material';
import Grid from '@mui/material/Grid';
import { styled } from '@mui/material/styles';
import Typography from '@mui/material/Typography';
import React from 'react';

const PREFIX = 'ListHeader';

const classes = {
  sectionHeader: `${PREFIX}-sectionHeader`,
};

const StyledGrid = styled(Grid)({
  [`&.${classes.sectionHeader}`]: {
    padding: '1em',
  },
});

export default function ListHeader(props: { title: string; actions?: React.ReactElement[] }) {
  const actions = props.actions || [];
  return (
    <StyledGrid
      container
      alignItems="flex-start"
      justifyContent="space-between"
      className={classes.sectionHeader}
    >
      {props.title && (
        <Grid>
          <Box p={0.5}>
            <Typography variant="h1">{props.title}</Typography>
          </Box>
        </Grid>
      )}
      {actions && actions.map((action, i: number) => <Grid key={i}>{action}</Grid>)}
    </StyledGrid>
  );
}
