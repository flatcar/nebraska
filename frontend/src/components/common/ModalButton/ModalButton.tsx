import AddIcon from '@mui/icons-material/Add';
import { IconButton } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';
import { useTranslation } from 'react-i18next';

import ApplicationEdit from '../../Applications/ApplicationEdit';
import ChannelEdit from '../../Channels/ChannelEdit';
import GroupEditDialog from '../../Groups/GroupEditDialog/GroupEditDialog';
import PackageEditDialog, {
  EditDialogProps as PackageEditDialogProps,
} from '../../Packages/EditDialog';

function ModalButton(props: { data: object; modalToOpen: string; icon?: string }) {
  const [showModal, setShowModal] = React.useState(false);
  const { t } = useTranslation();

  function close() {
    setShowModal(false);
  }

  function open() {
    setShowModal(true);
  }

  const options = {
    create: true,
    show: showModal,
    data: props.data,
    onHide: close,
  };

  let modal = null;
  switch (props.modalToOpen) {
    case 'AddApplicationModal':
      modal = <ApplicationEdit {...options} />;
      break;
    case 'AddGroupModal':
      modal = <GroupEditDialog {...options} />;
      break;
    case 'AddChannelModal':
      modal = <ChannelEdit {...options} />;
      break;
    case 'AddPackageModal':
      modal = <PackageEditDialog {...(options as PackageEditDialogProps)} />;
      break;
  }

  // @todo: verify whether aria-label should be more specific (in which
  // case it should be set from the caller).
  return (
    <div>
      <IconButton
        size="small"
        aria-label={t('frequent|add_lower')}
        onClick={open}
        data-testid="modal-button"
      >
        <AddIcon fontSize="large" sx={{ color: theme => theme.palette.titleColor }} />
      </IconButton>
      {modal}
    </div>
  );
}

ModalButton.propTypes = {
  modalToOpen: PropTypes.string.isRequired,
  data: PropTypes.object,
};

export default ModalButton;
