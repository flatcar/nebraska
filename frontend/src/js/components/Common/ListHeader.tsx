import { Box } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React from 'react';

const useStyles = makeStyles(theme => ({
  sectionHeader: {
    padding: '1em',
  },
}));

export default function ListHeader(props: {title: string; actions?: React.ReactElement[]}) {
  const classes = useStyles();
  const actions = props.actions || [];
  return (
    <Grid
      container
      alignItems="flex-start"
      justify="space-between"
      className={classes.sectionHeader}
    >
      {props.title &&
      <Grid item>
        <Box p={0.5}>
          <Typography variant="h4">{props.title}</Typography>
        </Box>
      </Grid>
      }
      {actions &&
              actions.map((action, i: number) =>
                <Grid item key={i}>
                  {action}
                </Grid>
              )
      }
    </Grid>
  );
}
