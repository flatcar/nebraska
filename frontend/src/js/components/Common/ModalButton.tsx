import { IconButton, makeStyles, useTheme } from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import PropTypes from 'prop-types';
import React from 'react';
import ApplicationEditDialog from '../Applications/EditDialog';
import ChannelEditDialog from '../Channels/EditDialog';
import GroupEditDialog from '../Groups/EditDialog';
import PackageEditDialog from '../Packages/EditDialog';

const useStyles = makeStyles((theme) => ({
  root: {
    color: theme.palette.titleColor
  }}));

function ModalButton(props: {data: object; modalToOpen: string; icon?: string}) {
  const [showModal, setShowModal] = React.useState(false);
  const classes = useStyles();

  function close() {
    setShowModal(false);
  }

  function open() {
    setShowModal(true);
  }

  const options = {
    create: true,
    show: showModal,
    data: props.data,
    onHide: close,
  };

  let modal = null;
  switch (props.modalToOpen) {
    case 'AddApplicationModal':
      modal = <ApplicationEditDialog {...options} />;
      break;
    case 'AddGroupModal':
      modal = <GroupEditDialog {...options} />;
      break;
    case 'AddChannelModal':
      modal = <ChannelEditDialog {...options} />;
      break;
    case 'AddPackageModal':
      modal = <PackageEditDialog {...options} />;
      break;
  }

  // @todo: verify whether aria-label should be more specific (in which
  // case it should be set from the caller).
  return (
    <div>
      <IconButton size="small" aria-label="add" onClick={open} data-testid="modal-button" >
        <AddIcon fontSize="large" className={classes.root}/>
      </IconButton>
      {modal}
    </div>
  );
}

ModalButton.propTypes = {
  modalToOpen: PropTypes.string.isRequired,
  data: PropTypes.object
};

export default ModalButton;
