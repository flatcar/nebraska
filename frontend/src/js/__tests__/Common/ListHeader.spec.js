import {render} from '@testing-library/react';
import React from 'react';
import ListHeader from '../../components/Common/ListHeader';
import ModalButton from '../../components/Common/ModalButton';

describe('List Header', () => {
  const minProps = {
    title: 'Applications',
    actions: [
      <ModalButton
        modalToOpen="AddApplicationModal"
        data={{applications: []}}
      />
    ]
  };
  it('should render correct List Header title', () => {
    const {getByText} = render(<ListHeader title={minProps.title}/>);
    expect(getByText(minProps.title)).toBeInTheDocument();
  });
  it('should render correct List Header actions', () => {
    const {asFragment} = render(<ListHeader actions={minProps.actions}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
