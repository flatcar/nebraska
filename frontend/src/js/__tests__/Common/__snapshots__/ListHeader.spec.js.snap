// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Header should render correct List Header actions 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-sectionHeader-137 MuiGrid-container MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-space-between"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div>
        <button
          aria-label="add"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
          data-testid="modal-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-root-241 MuiSvgIcon-fontSizeLarge"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
